src/item.cpp

```diff
@@ -4134,16 +4272,27 @@ item_location item::pick_reload_ammo( player &u, bool interactive ) const
     int i = 0;
     for( auto& e : ammo_list ) {
         const item *it = e.get_item();
-        const auto& ammo_def = *it->type->ammo;
-        std::string row = it->type->nname( it->charges ) + string_format( " (%d)", it->charges );
+        const auto curammo = it->ammo_data(); // nullptr for empty magazines
+
+        std::string row;
+        if( it->is_magazine() && curammo ) {
+            long qty = it->ammo_remaining();
+            row = string_format( "%s with %s (%d)", it->type->nname( 1 ).c_str(), curammo->nname( qty ).c_str(), qty );
```

Comment:
Add a comment starting with `~` in the line above the string literal, for example like 2cdb2f6419d99a13bbff5cfc494e949f8208ef74.